src/node/routes/login.ts

```diff
@@ -74,12 +74,13 @@ router.post("/", async (req, res) => {
 
     if (
       req.args["hashed-password"]
-        ? safeCompare(hash(req.body.password), req.args["hashed-password"])
+        ? isHashLegacyMatch(req.body.password, req.args["hashed-password"])
         : req.args.password && safeCompare(req.body.password, req.args.password)
     ) {
+      const hashedPassword = req.args["hashed-password"] ? hashLegacy(req.body.password) : hash(req.body.password)
```

Comment:
Don't we also have to use `isHashMatch` here too? I'm a little confused at what you're trying to do with both the if statement and the `const hashedPassword` variable - and we might need to take a second look at auth as a whole :sweat_smile:

Reply:
IIRC, `safeCompare` was used to compare the hashed-password with the `req.body.password`. We use `isHashLegacyMatch` which uses `safeCompare` inside it I believe. This is for backwards-compatibility.

`const hashedPassword` is needed for backwards-compatibility as well. If they're using the legacy `sha256` hashed-password, then we need to use that as `hashedPassword` which gets stored as a cookie. If they use the new formula then we use `hash`.

Hope that makes sense! Let me know how I can make that more clear.